In [38]:
import pandas as pd

# Read in the data
df = pd.read_csv('SKU.csv')

In [39]:
df.head()

sku_id  Product ID  Price  Discount  Stock Is Active  Is Stock  Weight  \
0      87          84    720       0.0      0      True      True     NaN   
1    3188         388   1875       0.0      0     False     False     NaN   
2    2838         388   3750       0.0      0     False      True     NaN   
3    2479          84    383       5.0      0      True     False     NaN   
4    2478          84    383       5.0      0      True     False     NaN   

   Unit  All Types  Is Deal  
0   NaN        NaN    False  
1   NaN        NaN    False  
2   NaN        NaN    False  
3   NaN        NaN     True  
4   NaN        NaN     True

In [40]:
df.shape

(3593, 11)

In [41]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3593 entries, 0 to 3592
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   sku_id      3593 non-null   int64  
 1   Product ID  3593 non-null   int64  
 2   Price       3593 non-null   int64  
 3   Discount    3593 non-null   float64
 4   Stock       3593 non-null   int64  
 5   Is Active   3592 non-null   object 
 6   Is Stock    3593 non-null   bool   
 7   Weight      0 non-null      float64
 8   Unit        0 non-null      float64
 9   All Types   0 non-null      float64
 10  Is Deal     3593 non-null   bool   
dtypes: bool(2), float64(4), int64(4), object(1)
memory usage: 259.8+ KB


In [42]:
df = df.drop(['Weight','Unit','All Types'], axis = 1)

In [43]:
df.head()

sku_id  Product ID  Price  Discount  Stock Is Active  Is Stock  Is Deal
0      87          84    720       0.0      0      True      True    False
1    3188         388   1875       0.0      0     False     False    False
2    2838         388   3750       0.0      0     False      True    False
3    2479          84    383       5.0      0      True     False     True
4    2478          84    383       5.0      0      True     False     True

In [48]:
df['sku_id'].value_counts()

87      1
1796    1
3488    1
1741    1
2582    1
       ..
1378    1
1746    1
1752    1
1753    1
1637    1
Name: sku_id, Length: 3593, dtype: int64

In [44]:
df['Product ID'].value_counts()

51     119
326     91
777     80
839     75
370     67
      ... 
451      1
323      1
351      1
510      1
804      1
Name: Product ID, Length: 491, dtype: int64

In [47]:
df[df['Product ID'] == 51]

sku_id  Product ID  Price  Discount  Stock Is Active  Is Stock  Is Deal
30       142          51     96       0.0      0      True      True    False
32       147          51     96       0.0      0      True      True    False
33       146          51     96       0.0      0      True      True    False
37       311          51     75       0.0      0     False      True    False
70       322          51    300       0.0      0     False      True    False
...      ...         ...    ...       ...    ...       ...       ...      ...
3046     340          51     99       0.0      0     False      True    False
3048    2314          51     87       0.0      0     False      True    False
3119     315          51    188       0.0      0     False      True    False
3120     353          51    188       0.0      0     False      True    False
3186    2317          51    458       0.0     38     False      True    False

[119 rows x 8 columns]

In [21]:
df1 = pd.read_csv('Order Items.csv')

In [22]:
df1.head()

order_id  sku_id  price  discount  qty  order_status_id
0    153088     288   1005         0    1              4.0
1    153088     282    593         0    3              4.0
2    153088    1401   1223         0    2              4.0
3    151396     452   1530         0    1              4.0
4    153101     452   1538         0    1              6.0

In [23]:
df1.shape

(134561, 6)

In [24]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 134561 entries, 0 to 134560
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   order_id         134561 non-null  int64  
 1   sku_id           134561 non-null  int64  
 2   price            134561 non-null  int64  
 3   discount         134561 non-null  int64  
 4   qty              134561 non-null  int64  
 5   order_status_id  134529 non-null  float64
dtypes: float64(1), int64(5)
memory usage: 6.2 MB


In [25]:
df1['sku_id'].value_counts()

452     2863
415     2255
655     2022
93      1971
285     1946
        ... 
722        1
3378       1
2661       1
2655       1
3217       1
Name: sku_id, Length: 1416, dtype: int64

In [ ]:
df1['sku_id'].value_counts()


- we find the number of unique values in each column of the dataset.
- According to our order items data,  we find that SKU ID `452` has the highest number of order items.
- 

In [28]:
df.head(), df1.head()

(   sku_id  Product ID  Price  Discount  Stock Is Active  Is Stock  Is Deal
 0      87          84    720       0.0      0      True      True    False
 1    3188         388   1875       0.0      0     False     False    False
 2    2838         388   3750       0.0      0     False      True    False
 3    2479          84    383       5.0      0      True     False     True
 4    2478          84    383       5.0      0      True     False     True,
    order_id  sku_id  price  discount  qty  order_status_id
 0    153088     288   1005         0    1              4.0
 1    153088     282    593         0    3              4.0
 2    153088    1401   1223         0    2              4.0
 3    151396     452   1530         0    1              4.0
 4    153101     452   1538         0    1              6.0)

In [32]:
df[df['sku_id'] == 452]

sku_id  Product ID  Price  Discount  Stock Is Active  Is Stock  Is Deal
3426     452         774   1620       0.0   1886      True      True    False

In [49]:
produc_cat = pd.read_csv('Product Categories.csv')

In [50]:
produc_cat.head()

Category ID  Product ID
0           10         292
1           53         292
2           11           3
3           62           3
4            7           4

In [54]:
produc_cat.shape

(1772, 2)

In [53]:
produc_cat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1772 entries, 0 to 1771
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   Category ID  1772 non-null   int64
 1   Product ID   1772 non-null   int64
dtypes: int64(2)
memory usage: 27.8 KB


In [52]:
produc_cat[produc_cat['Product ID'] == 51]

Category ID  Product ID
1275          164          51
1276          172          51

In [56]:
produc_cat['Product ID'].value_counts()

292    2
587    2
582    2
626    2
627    2
      ..
90     2
378    2
388    2
217    1
25     1
Name: Product ID, Length: 887, dtype: int64

In [57]:
df = pd.read_csv("Sales Agents Orders.csv")

In [58]:
df.head()

id  order_id  agent_user_id
0  6715    188049          20773
1  5191    176219          47070
2  4367    167133          46908
3  4385    167219          46908
4  4431    167419          44736

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4141 entries, 0 to 4140
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   id             4141 non-null   int64
 1   order_id       4141 non-null   int64
 2   agent_user_id  4141 non-null   int64
dtypes: int64(3)
memory usage: 97.2 KB


In [72]:
df['agent_user_id'].value_counts().sort_values(ascending = False).head(5)

20282    348
18871    277
45342    276
16781    275
37402    211
Name: agent_user_id, dtype: int64

# agent_user_id 348 has the highest number of order items.
- top sellers are selected for team lead
- 


In [70]:
x = df.sort_values(['order_id'], ascending=False).groupby('agent_user_id')
x

In [63]:
df.head()

id  order_id  agent_user_id
0  6715    188049          20773
1  5191    176219          47070
2  4367    167133          46908
3  4385    167219          46908
4  4431    167419          44736